src/Shared/DiagnosticSourceInstrumentation/PropertyFetcher.cs

```diff
@@ -85,54 +73,109 @@ public bool TryFetch(object obj, out T value, bool skipObjNullCheck = false)
     }
 
     // see https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
+#if NET6_0_OR_GREATER
+    [RequiresUnreferencedCode(TrimCompatibilityMessage)]
+#endif
     private class PropertyFetch
     {
-        public static PropertyFetch Create(TypeInfo type, string propertyName)
+        public static PropertyFetch? Create(TypeInfo type, string propertyName)
         {
-            var property = type.DeclaredProperties.FirstOrDefault(p => string.Equals(p.Name, propertyName, StringComparison.OrdinalIgnoreCase));
-            if (property == null)
-            {
-                property = type.GetProperty(propertyName);
-            }
-
-            return CreateFetcherForProperty(property);
+            var property = type.DeclaredProperties.FirstOrDefault(p => string.Equals(p.Name, propertyName, StringComparison.OrdinalIgnoreCase)) ?? type.GetProperty(propertyName);
+            return CreateFetcherForProperty(property!);
```

Comment:
```suggestion
            return CreateFetcherForProperty(property);
```

See below for how you can remove this.